Add version note (#17432)
docker/docs: scout/dashboard.md

- The most recent version of the image and the vulnerabilities for that version. Clicking the link for the base image opens [the image layer view](#image-layer-view).
- The operating system and architecture of the image.
- The base image and version used by the repository and the vulnerabilities for that version. Clicking the link for the base image opens [the image layer view](#image-layer-view).

  > **Note**
  >

>   > Docker Scout detects the base image used by a repository by using the digest hash. However, if the hash is linked to different tags, Docker Scout can't show the precise version.

Reviewer: ?

```suggestion
  > Docker Scout detects the base image used by a repository by using the digest hash. However, if the hash is associated with more than one tag, Docker Scout can't show the precise version.
```

Author: @dvdksn Thanks, I was stuck on that particular word for a while, perfect.

>   > Docker Scout detects the base image used by a repository by using the digest hash. However, if the hash is linked to different tags, Docker Scout can't show the precise version.

Author: @eunomie Fair point, I think at the time I was trying to differentiate two different applications of the term.

- The most recent version of the image and the vulnerabilities for that version. Clicking the link for the base image opens [the image layer view](#image-layer-view).
- The operating system and architecture of the image.
- The base image and version used by the repository and the vulnerabilities for that version. Clicking the link for the base image opens [the image layer view](#image-layer-view).

  > **Note**
  >

>   > Docker Scout detects the base image of an image used in a repository by matching layer `diff_ids`.

Reviewer 2: I feel like diff IDs are a bit of an implementation detail - we should obscure that complexity if possible. Maybe just this?

(also suggested removing "in a repository", since this is true of exclusively local images as well)

```suggestion
  > Docker Scout detects the base image of an image by matching layer IDs.
```

Author: I like this suggestion. @cdupuis Is this enough clarity/accuracy compromise?

Reviewer 3: Don't think this is technically really correct but it shouldn't matter much. As a reader I wonder what a `layer ID` is. What about just:

```suggestion
  > Docker Scout detects the base image of an image by matching layer content hashes.
```

>   > However, it's possible for different images and tags to use these layers. In these cases, Docker Scout's

Reviewer 2: ```suggestion
  > However, it's possible for multiple tags to point at the same layers. In these cases, Docker Scout's
```

Author: I took bits of this into a change

>   > base image detection might not be precise and potentially return a different tag to the one used in the Dockerfile.

Reviewer 2: Maybe it's just me, but my eyes stumbled a bit over parsing the scope of the _not_ when I first read this. The first half of the conjunction is essentially a subset of the second half, so we could just skip it?

```suggestion
  > base image detection could return a different tag to the one used in the Dockerfile.
```
